examples/userapi/05_conditional_dimension.ipynb

```diff
@@ -0,0 +1,851 @@
+      "    to guard against out-of-bounds accesses.\n",
+      "\n",
+      "    >>> from sympy import And\n",
+      "    >>> ci = ConditionalDimension(name='ci', parent=i,\n",
+      "    ...                           condition=And(g[i] > 0, g[i] < 4, evaluate=False))\n",
+      "    >>> f = Function(name='f', shape=(int(size/factor),), dimensions=(ci,))\n",
+      "    >>> op = Operator(Eq(f[g[i]], f[g[i]] + 1))\n",
+      "\n",
+      "    The Operator generates the following for-loop (pseudocode)\n",
+      "\n",
+      "    .. code-block:: C\n",
+      "\n",
+      "        for (int i = i_m; i <= i_M; i += 1) {\n",
+      "          if (g[i] > 0 && g[i] < 4) {\n",
+      "            f[g[i]] = f[g[i]] + 1;\n",
+      "          }\n",
+      "        }\n",
+      "\n",
+      "    \n"
+     ]
+    }
+   ],
+   "source": [
+    "from devito import ConditionalDimension\n",
+    "print(ConditionalDimension.__doc__)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Use case A (Honour a certain condition)\n",
+    "\n",
+    "\n",
+    "## Defining a condition\n",
+    "Now, it's time to show a more descriptive example. Let's define a conditional that updates only the values of the function that are larger than 0, f(x,y)>0. We name our Conditional Dimension `cy` (conditional y).\n",
+    "In this example we want to update again by one all the elements in `f` that are larger than 0, $f(x,y) > 0$\n",
+    "Before updating the elements we initialize our data again to the eye function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Operator `Kernel` run in 0.01 s\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "Data([[2., 1., 1., 1., 1., 1., 1., 1., 1., 1.],\n",
+       "      [1., 2., 1., 1., 1., 1., 1., 1., 1., 1.],\n",
+       "      [1., 1., 2., 1., 1., 1., 1., 1., 1., 1.],\n",
+       "      [1., 1., 1., 2., 1., 1., 1., 1., 1., 1.],\n",
+       "      [1., 1., 1., 1., 2., 1., 1., 1., 1., 1.],\n",
+       "      [1., 1., 1., 1., 1., 2., 1., 1., 1., 1.],\n",
+       "      [1., 1., 1., 1., 1., 1., 2., 1., 1., 1.],\n",
+       "      [1., 1., 1., 1., 1., 1., 1., 2., 1., 1.],\n",
+       "      [1., 1., 1., 1., 1., 1., 1., 1., 2., 1.],\n",
+       "      [1., 1., 1., 1., 1., 1., 1., 1., 1., 2.]], dtype=float32)"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "#NBVAL_IGNORE_OUTPUT\n",
+    "\n",
+    "from devito.types import Array, Lt, Le, Gt, Ge, Ne\n",
+    "\n",
+    "f.data[:, :] = np.eye(10)\n",
+    "\n",
+    "condition = Gt(f, 0)\n",
+    "\n",
+    "ci = ConditionalDimension(name='ci', parent=y, condition=condition)\n",
+    "\n",
+    "op = Operator(Eq(f, f + 1))\n",
```

Comment:
Not sure from this point onwards is needed in the current block - it's repetition, can just go straight to proper case in the next box.

Also, worth adding some comments where necessary, i.e. above the line `condition = Gt(f, 0)`